src/functions/tarsUpload/domain/util/TestTypeLookup.ts

```diff
@@ -0,0 +1,24 @@
+export const determineDl25TestType = (category: string): number => {
+    // DL25 test category to test type mapping, as per the TARS TEST_CATEGORY_CROSS_REFERENCE table,
+     // documented at https://wiki.i-env.net/display/MES/Test+Category+Cross+Reference and agreed with DVSA MI Team.
+  const mapping: Map<string, number> = new Map([
+       ['ADI2', 10],
+       ['B', 2], ['B+E', 2],
+       ['C', 3], ['C+E', 3], ['C1', 3], ['C1+E', 3],
+       ['D', 4], ['D+E', 4], ['D1', 4], ['D1+E', 4],
+       ['F', 5],
+       ['G', 6],
+       ['H', 7],
+       ['K', 8],
+       // Note that some extra data will be needed in MES to indentify CPC tests, if MES adds support for them...
+       // LGV (Lorry) CPC (all C Categories) => 44
+       // PCV (Bus) CPC (all D Categories) => 44
+       ['A1M1', 16], ['A1M2', 1],
+       ['A2M1', 16], ['A2M2', 1],
+       ['AM1', 16], ['AM2', 1],
+       ['AMM1', 17], ['AMM2', 9],
+  ]);
+  const testType = mapping.get(category);
+       // Note if no mapping exists will default to category B value of 2
+  return (testType === undefined) ? 2  : testType;
```

Comment:
Are we certain that this is the way how we want to deal with 'no mapping' situation?
Obviously it "will work" but sending the value of 2 as default might not be the "right thing to do" here IMHO... 
It seems to me that this is something like a data problem (with a request) when there is no appropriate mapping - BAD REQUEST-ish so compensating for it with a valid default might just result in wrong licence type issued by DVLA...

I'd suggest throwing an error here and dealing with it with a 400 HTTP code - this should mark the test result in ERROR state which could be potentially mitigated on DES side of things  instead of TARS / DVLA ;}

Reply:
Okay. will look in to changing this. need to consider the full implications - as it is called from at least one other service.

Reply:
amended to return undefined if it doesn't find the category in the map - but  created a new exception to indicate problem with category.

Reply:
Looks, good - thanks :+1:
You haven't removed that "note" comment  which is not needed anymore ;}
Can we get a test covering 400 code staff? now sure if we have coverage there